shared_code/utils.py

```diff
@@ -145,3 +146,40 @@ def locate_potential_duplicate(name, zipcode, client, collection):
     if dupe_candidate is not None:
         return dupe_candidate["name"]
     return False
+
+
+def validate_data(df):
+    logging.basicConfig(level=logging.DEBUG)
+    logger = logging.getLogger(__name__)
+
+    # Check if all the required columns ('name', 'address1', 'city', 'state', 'zip', 'serviceSummary')
+    # are in the dataframe
+    requiredColumns = ['name', 'address1', 'city', 'state', 'zip', 'serviceSummary']
+    missingColumns = []
+    for column in requiredColumns:
+        if not column in df.columns:
+            missingColumns.append(column)
+    if len(missingColumns) > 0:
+        raise Exception('The data frame does not contain the following column(s): ' + str(missingColumns))
+
+    rows_to_drop = set()
+
+    # Identifies any row that contains a null or 'NONE' value in the 'name', 'address1', or 'serviceSummary' columns as
+    # invalid
+    for column in ['name', 'address1', 'serviceSummary']:
+        df_none = df[(df[column].isna()) | (df[column] == '') | (df[column].str.upper() == 'NONE')].index.values
```

Comment:
I decided to make a function called `empty_slots` that returns all the empty slots inside the data frame. This can be used for validating both individual columns like `name`, `address1`, and `serviceSummary` as well as groups of columns like `city`, `state`, and `zip`.

The code for `empty_slots` is:
```
def empty_slots(df, column):
     return (df[column].isna()) | (df[column] == '') | (df[column].str.upper() == 'NONE')
```